engines/private/private.cpp

```diff
@@ -0,0 +1,1086 @@
+}
+
+void PrivateEngine::restartGame() {
+    debugC(1, kPrivateDebugFunction, "restartGame");
+
+    for (NameList::iterator it = maps.variableList.begin(); it != maps.variableList.end(); ++it) {
+        Private::Symbol *sym = maps.variables.getVal(*it);
+        if (*(sym->name) != "kAlternateGame")
+            sym->u.val = 0;
+    }
+
+    // Diary
+    for (NameList::iterator it = maps.locationList.begin(); it != maps.locationList.end(); ++it) {
+        Private::Symbol *sym = maps.locations.getVal(*it);
+        sym->u.val = 0;
+    }
+    inventory.clear();
+    _dossiers.clear();
+
+    // Sounds
+    _AMRadio.clear();
+    _policeRadio.clear();
+    _phone.clear();
+    _playedPhoneClips.clear();
+
+    // Movies
+    _repeatedMovieExit = "";
+    _playedMovies.clear();
+
+}
+
+Common::Error PrivateEngine::loadGameStream(Common::SeekableReadStream *stream) {
+    // We don't want to continue with any sound from a previous game
+    stopSound(true);
+
+    Common::Serializer s(stream, nullptr);
+    debugC(1, kPrivateDebugFunction, "loadGameStream");
+    setNextSetting(new Common::String(kStartGame));
+    int val;
+
+    for (NameList::iterator it = maps.variableList.begin(); it != maps.variableList.end(); ++it) {
+        s.syncAsUint32LE(val);
+        Private::Symbol *sym = maps.variables.getVal(*it);
+        sym->u.val = val;
+    }
+
+    // Diary
+    for (NameList::iterator it = maps.locationList.begin(); it != maps.locationList.end(); ++it) {
+        s.syncAsUint32LE(val);
+        Private::Symbol *sym = maps.locations.getVal(*it);
+        sym->u.val = val;
+    }
+    uint32 size = stream->readUint32LE();
+    for (uint32 i = 0; i < size; ++i) {
+        inventory.push_back(stream->readString());
+    }
+
+    // Dossiers
+    size = stream->readUint32LE();
+    Common::String *file = NULL;
+    for (uint32 i = 0; i < size; ++i) {
+        file = new Common::String(stream->readString());
+
+        DossierInfo *m = (DossierInfo *)malloc(sizeof(DossierInfo));
+        m->page1 = file;
+
+        file = new Common::String(stream->readString());
+        if (file->size() == 0) {
+            m->page2 = NULL;
+        } else {
+            m->page2 = file;
+        }
+        _dossiers.push_back(*m);
+    }
+
+    // Radios
+    size = stream->readUint32LE();
+    _AMRadio.clear();
+
+    for (uint32 i = 0; i < size; ++i) {
+        _AMRadio.push_back(stream->readString());
+    }
+
+    size = stream->readUint32LE();
+    _policeRadio.clear();
+
+    for (uint32 i = 0; i < size; ++i) {
+        _policeRadio.push_back(stream->readString());
+    }
+
+    size = stream->readUint32LE();
+    _phone.clear();
+
+    for (uint32 j = 0; j < size; ++j) {
+        PhoneInfo *i = (PhoneInfo *)malloc(sizeof(PhoneInfo));
+
+        i->sound = new Common::String(stream->readString());
+        i->flag  = maps.variables.getVal(stream->readString());
+        i->val   = stream->readUint32LE();
+        _phone.push_back(*i);
```

Comment:
Again, you're leaking `PhoneInfo` here. You're pushing an object (`*i`), which effectively copies it and on next loop iteration overwrite the allocated memory chunk pointed to by `i`. Thus, you will leak `size` of `PhoneInfo`'s

Reply:
Fixed.